src/button-dropdown/__tests__/button-dropdown.test.tsx

```diff
@@ -317,3 +318,57 @@ describe('with main action', () => {
     expect(wrapper.findMainAction()!.getElement()).toHaveFocus();
   });
 });
+
+test('should work in controlled context', () => {
+  const onClickSpy = jest.fn();
+  function StateWrapper() {
+    const [checked, setChecked] = useState(true);
+    return (
+      <ButtonDropdown
+        items={[{ id: 'id', text: 'option', checked: checked, itemType: 'checkbox' }]}
+        onItemClick={event => {
+          onClickSpy(event.detail);
+          setChecked(event.detail.checked!);
+        }}
+      />
+    );
+  }
+
+  const { container } = render(<StateWrapper />);
+  const wrapper = createWrapper(container).findButtonDropdown()!;
+
+  // Click the item and verify it called the onClickSpy with the correct state and has correct value
+  wrapper.openDropdown();
+  expect(wrapper.findItemValueById('id')).toBe('true');
```

Comment:
Can we call the method `findItemCheckedById()` - this aligns with how the checked state is set in HTML. It took me a minute to get why the value is expected to be 'true'.

Reply:
yeah that makes sense, thanks for the suggestion.